### PR TITLE
kafka-resumption tests: Sinks were not tested through toxiproxy

### PR DESCRIPTION
**Pull request by def-**
### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Review by nrainer-materialize — APPROVED** [review]
Nice!

How did you spot this?

**Comment by def-** [comment]
I was trying to cover https://github.com/MaterializeInc/materialize/issues/25833 and had a look at this test case for it.

**Comment by def-** [comment]
@petrosagg This now seems to fail with the sink-queue-full case (`bin/mzcompose --find kafka-resumption down && bin/mzcompose --find kafka-resumption run sink-queue-full`) and never recovers:
```
[["2024-03-07 15:33:25.866 UTC", "u6", "stalled", "kafka: Transaction error: Failed to initialize Producer ID: Local: Timed out: Failed to initialize Producer ID: Local: Timed out", "{\"hints\":[\"If you're running a single Kafka broker, ensure that the configs transaction.state.log.replication.factor, transaction.state.log.min.isr, and offsets.topic.replication.factor are set to 1 on the broker\"],\"namespaced\":{\"kafka\":\"Transaction error: Failed to initialize Producer ID: Local: Timed out: Failed to initialize Producer ID: Local: Timed out\"}}"], ["2024-03-07 15:35:05.993 UTC", "u12", "stalled", "kafka: Transaction error: Failed to initialize Producer ID: Local: Timed out: Failed to initialize Producer ID: Local: Timed out", "{\"hints\":[\"If you're running a single Kafka broker, ensure that the configs transaction.state.log.replication.factor, transaction.state.log.min.isr, and offsets.topic.replication.factor are set to 1 on the broker\"],\"namespaced\":{\"kafka\":\"Transaction error: Failed to initialize Producer ID: Local: Timed out: Failed to initialize Producer ID: Local: Timed out\"}}"], ["2024-03-07 15:36:45.205 UTC", "u18", "stalled", "kafka: Transaction error: Failed to initialize Producer ID: Local: Timed out: Failed to initialize Producer ID: Local: Timed out", "{\"hints\":[\"If you're running a single Kafka broker, ensure that the configs transaction.state.log.replication.factor, transaction.state.log.min.isr, and offsets.topic.replication.factor are set to 1 on the broker\"],\"namespaced\":{\"kafka\":\"Transaction error: Failed to initialize Producer ID: Local: Timed out: Failed to initialize Producer ID: Local: Timed out\"}}"], ["2024-03-07 15:38:25.165 UTC", "u24", "stalled", "kafka: Transaction error: Failed to initialize Producer ID: Local: Timed out: Failed to initialize Producer ID: Local: Timed out", "{\"hints\":[\"If you're running a single Kafka broker, ensure that the configs transaction.state.log.replication.factor, transaction.state.log.min.isr, and offsets.topic.replication.factor are set to 1 on the broker\"],\"namespaced\":{\"kafka\":\"Transaction error: Failed to initialize Producer ID: Local: Timed out: Failed to initialize Producer ID: Local: Timed out\"}}"], ["2024-03-07 15:39:28.784 UTC", "u30", "stalled", "kafka: Error creating topic output-byo-sink-145350 for sink: Admin operation error: OperationTimedOut (Local: Timed out)", "{\"namespaced\":{\"kafka\":\"Error creating topic output-byo-sink-145350 for sink: Admin operation error: OperationTimedOut (Local: Timed out)\"}}"]]
```
Is that expected when we have too much data in a sink while it has connection problems?

File: test/kafka-resumption/sink-queue-full/setup.td [hunk 1]
```diff
@@ -18,14 +18,20 @@ $ kafka-create-topic topic=largeinput
 $ kafka-ingest topic=largeinput format=bytes
 A,B,0
 
-> CREATE CONNECTION IF NOT EXISTS kafka_conn TO KAFKA (BROKER '${testdrive.kafka-addr}', SECURITY PROTOCOL PLAINTEXT);
+> CREATE CONNECTION IF NOT EXISTS kafka_conn1 TO KAFKA (BROKER '${testdrive.kafka-addr}', SECURITY PROTOCOL PLAINTEXT);
+> CREATE CONNECTION IF NOT EXISTS kafka_conn2 TO KAFKA (BROKER 'toxiproxy:9093', SECURITY PROTOCOL PLAINTEXT);
 
 # The source intentionally does not go through toxiproxy.
```
Comment (petrosagg) [inline test/kafka-resumption/sink-queue-full/setup.td:24]: @def- I think this test needs to remain as is according to this comment

Comment (nrainer-materialize) [reply test/kafka-resumption/sink-queue-full/setup.td:24]: We could make this more obvious by moving the comment above the `CREATE CONNECTION` 😀

**Review by nrainer-materialize — APPROVED** [review]

**Pull request merged**